driver-core/src/test/java/com/datastax/driver/core/TypeCodecNestedUDTAndTupleIntegrationTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.*;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.datastax.driver.core.utils.CassandraVersion;
+
+/**
+ * This tests that the CodecRegistry is able to create codecs on the fly
+ * for nested UDTs and tuples received from a ResultSet, i.e. the first time
+ * the registry encounters the UDT or tuple, it's coming from a ResultSet,
+ * not from the Cluster's metadata.
+ *
+ * @jira_ticket JAVA-847
+ */
+@CassandraVersion(major = 2.1)
+public class TypeCodecNestedUDTAndTupleIntegrationTest extends CCMBridge.PerClassSingleNodeCluster {
+
+    @Override
+    protected Collection<String> getTableDefinitions() {
```

Comment:
Probably good to have a test for a tuple inside a UDT and a UDT inside a tuple, also 3 layers deep of something.